modules/snooker/src/main/ts/ephox/snooker/api/CopySelected.ts

```diff
@@ -105,39 +112,24 @@ const clean = (replica: SugarElement<HTMLTableElement>, stats: StatsStruct, widt
     });
   }
 
-  const replicaTableSize = TableSize.getTableSize(replica);
-  replicaTableSize.adjustTableWidth(widthDelta);
-};
-
-const getTableWidthDelta = (tableSize: TableSize, stats: StatsStruct): number => {
-  /*
-    Calculate new width by comparing width of selected columns to
-    width of all columns, which is the ratio to apply to the full table width.
+  const tableSize = TableSize.getTableSize(replica);
+  tableSize.adjustTableWidth(widthDelta);
 
-    We have to do this due to padding/margin etc.
-    (e.g. a 1000px wide table might have 3 x 320px wide columns)
-
-    Note: We will be off by a few pixels due to borders
-  */
-  const getColsWidth = (cols: DetailExt[]) => {
-    return cols.reduce((acc, col) => {
-      return acc + Width.getOuter(col.element);
-    }, 0);
-  };
-
-  const uniqueCols = ColUtils.uniqueColumns(stats.allCells);
-  const uniqueSelectedCols = ColUtils.uniqueColumns(stats.selectedCells);
+  // TODO: If using relative widths, ensure cell and column widths are redistributed
+};
 
+const getTableWidthDelta = (table: SugarElement<HTMLTableElement>, warehouse: Warehouse, tableSize: TableSize, stats: StatsStruct): number => {
   // short circuit entire table selected
-  if (uniqueSelectedCols.length === uniqueCols.length) {
+  if (stats.minCol === 0 && warehouse.grid.columns === stats.maxCol + 1) {
     return 0;
   }
 
-  const allColsWidth = getColsWidth(uniqueCols);
-  const selectedColsWidth = getColsWidth(uniqueSelectedCols);
-
+  const colWidths = ColumnSizes.getPixelWidths(warehouse, table, tableSize);
+  const allColsWidth = Arr.foldl(colWidths, (acc, width) => acc + width, 0);
+  const selectedColsWidth = Arr.foldl(colWidths.slice(stats.minCol, stats.maxCol + 1), (acc, width) => acc + width, 0);
```

Comment:
Do we have an `Arr.sum`? Should we?

Reply:
As far as I can tell there isn't an `Arr.sum`, I'm not opposed to implementing - although probably out of scope for this PR